FT Assignments API app

mugiwaramunyi opened the pull request:
**What does this PR do?**

This adds an Assignments API app to the Django project CitizenReporter_API. This API is composed of two endpoints:
-/assignments
-/assignments/<id>

**Description of Task to be completed?**

Create an API endpoint to create, update, delete and return a list of current assignments. This endpoint should be publicly available to anyone

**How should this be manually tested?**

`python manage.py test assignments.tests`


Any background context you want to provide?

This project is being re-engineered from scratch, therefore all code added to this project is new.

mugiwaramunyi committed:
FEATURE new app assignments added to project

mugiwaramunyi committed:
Set up the settings and urls for the assignments app

mugiwaramunyi committed:
Models for assignments app as prescribed by TTL

mugiwaramunyi committed:
Serializers for assignments models using django_rest

mugiwaramunyi committed:
Views for assignments app, allows user to view one as well as all assignments

mugiwaramunyi committed:
Urls for the assignments app

mugiwaramunyi committed:
CRUD test suite for assignment app

mugiwaramunyi committed:
Add default json setting for REST_FRAMEWORK

mugiwaramunyi committed:
Allowing blank values for some fields in models

mugiwaramunyi committed:
Naming the urls to comply better with REST

mugiwaramunyi committed:
Adding tests to the app for basic CRUD operations

mugiwaramunyi committed:
Add migrations and db to gitignore

mugiwaramunyi committed:
Added the author and location fields to the model

mugiwaramunyi committed:
Added the author and locations fields to the serializer

mugiwaramunyi committed:
Refactors the test to accomodate the new author and location variables.

mugiwaramunyi committed:
Merge branch 'master' into ft_assignments_api_app

phillipahereza commented:
I see in your tests, when creating an assignment the post data includes the `id`, the ID is supposed to be automatically generated by the django ORM when it creates an entry in the database for that assignment

Jonathan-Ngondi commented:
Noted. I also thought about this when writing my code, but I forgot to seek
further clarification. Will refactor tonight.

On 19 Jul 2017 3:57 PM, "Phillip Ahereza" <notifications@github.com> wrote:

> I see in your tests, when creating an assignment the post data includes
> the id, the ID is supposed to be automatically generated by the django
> ORM when it creates an entry in the database for that assignment
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/CodeForAfricaLabs/CitizenReporter.API/pull/1#issuecomment-316377397>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AZEOIKj0d7GPeb3TDWC2aeldNRJMyGIwks5sPf00gaJpZM4OcrqI>
> .
>

phillipahereza commented:
@Jonathan-Ngondi could you also test the assignments featured image upload. Mock it 
https://swapps.io/blog/testing-files-with-pythondjango/

Jonathan-Ngondi commented:
Okay will try it out.

mugiwaramunyi committed:
Changed the app installment from assignments.app to just assignments

mugiwaramunyi committed:
Added the media_upload field and removed id from the model

mugiwaramunyi committed:
Adds the media_upload to the Serializer fields

mugiwaramunyi committed:
Tests with 1 failing test for image upload

mugiwaramunyi committed:
Merge branch 'ft_assignments_api_app' of https://github.com/Jonathan-Ngondi/CitizenReporter.API into ft_assignments_api_app

phillipahereza commented on assignments/views.py:
```diff
@@ -0,0 +1,18 @@
+from assignments.models import Assignment
```
run `isort -rc --atomic -y .` to sort your imports as specified in the `PEP8` standard

mugiwaramunyi committed:
Add uploads folder to gitignore

mugiwaramunyi committed:
Pep8 formatting

mugiwaramunyi committed:
-m
multiple variable changes
- MEDIA_CHOICES changed to accomodate more than one value
- scramble_uploaded_filename method added to prevent naming conflicts on uploaded files
- media_upload changed to ImageField from FileField
- num_of_responses has null=True argument added due to its read-only status

mugiwaramunyi committed:
Add read_only_field, num_of_responses

mugiwaramunyi committed:
Changes media_upload variable to featured_image

mugiwaramunyi committed:
Change media_upload field to featured_image

mugiwaramunyi committed:
Commented out the test_upload_picture_method

mugiwaramunyi committed:
Pep 8 formatting

mugiwaramunyi committed:
Adds nose, olefile, and Pillow to requirements

mugiwaramunyi committed:
Pep8 formmatting

mugiwaramunyi committed:
Adds .DS_Store to gitignore

mugiwaramunyi committed:
Minor documentation and code cleanup

mugiwaramunyi committed:
Minor documentation and code cleanup
